### PR TITLE
docs: speculations rules API javascript plugin

### PR DESCRIPTION
**Pull request by BrocksiNet**
Add docs about the new experimental javascript plugin.

### CODE REVIEW

**Review by shyim — APPROVED** [review]

File: guides/hosting/performance/performance-tweaks.md [hunk 1]
```diff
@@ -282,3 +282,18 @@ Disadvantages are:
...
+This is available starting with Shopware 6.6.10.0
+:::
+
+You can enable that **experimental feature** via `Admin > Settings > System > Storefront`. The JavaScript Plugin will then
+check if the [Browser supports the Speculation Rules API](https://caniuse.com/mdn-http_headers_speculation-rules) and if so,
+it will add a script tag to the head of the document. For the [eagerness option](https://developer.chrome.com/docs/web-platform/prerender-pages#eagerness)
+we are using `moderate` everywhere. That means a user must interact with a link to execute the pre-rendering.
+
+::: info
+Keep in mind that pre-rendering is putting extra load on your server and also can affect your [Analytics](https://developer.chrome.com/docs/web-platform/prerender-pages#impact-on-analytics).
```
Comment (Isengo1989) [inline guides/hosting/performance/performance-tweaks.md:301]:
```suggestion
Keep in mind that pre-rendering puts extra load on your server and can also affect your [Analytics](https://developer.chrome.com/docs/web-platform/prerender-pages#impact-on-analytics).
```

File: guides/hosting/performance/performance-tweaks.md [hunk 2]
```diff
@@ -282,3 +282,18 @@ Disadvantages are:
 To disable the Product Stream Indexer, you can set the following configuration:
 
 <<< @/docs/snippets/config/product_stream.yaml
+
+## Enable the Speculation Rules API
+
+::: info
+This is available starting with Shopware 6.6.10.0
```
Comment (Isengo1989) [inline guides/hosting/performance/performance-tweaks.md:289]:
```suggestion
This feature is available starting with Shopware 6.6.10.0
```

**Review by Isengo1989 — CHANGES_REQUESTED** [review]
Thx for the PR @BrocksiNet :tada: 

Added some small suggestions and a short explanation of what the feature does.

**Review by Isengo1989 — APPROVED** [review]
We will merge it as soon as the version is out. With this PR we are getting informed proactively by a comment as soon as it happens

**Comment by BrocksiNet** [comment]
Merged here: https://github.com/shopware/shopware/commit/2e7c627cf5a8a2252f3ec61086d29b47aa518812.

**Pull request merged**